### DISCUSSION TITLE
Strategy to get the language

### DISCUSSION THREAD

**Discussion by sawa-ko** [vinissimus/next-translate#1120] [Ideas]
Is there any possibility of having another strategy to obtain the language of the application other than creating a [lang] folder within the app folder? For example, get the language based on the cookie stored in the browser, since the nextjs server can read the cookies when pre-rendering the components, right?

**Comment by aralroca** [comment]
probably you can do it using a middleware and setting the `lang` parameter to the URLs, we also support the `lang` param for these scenarios. `/page?lang="en"`
